pkg/tso/keyspace_group_manager.go

```diff
@@ -257,20 +249,77 @@ func NewKeyspaceGroupManager(
 
 // Initialize this KeyspaceGroupManager
 func (kgm *KeyspaceGroupManager) Initialize() error {
-	// Load the initial keyspace group assignment from storage with time limit
-	done := make(chan struct{}, 1)
-	ctx, cancel := context.WithCancel(kgm.ctx)
-	go kgm.checkInitProgress(ctx, cancel, done)
-	watchStartRevision, defaultKGConfigured, err := kgm.initAssignment(ctx)
-	done <- struct{}{}
-	if err != nil {
+	rootPath := kgm.legacySvcRootPath
+	startKey := strings.Join([]string{rootPath, endpoint.KeyspaceGroupIDPath(mcsutils.DefaultKeyspaceGroupID)}, "/")
+	endKey := strings.Join(
+		[]string{rootPath, clientv3.GetPrefixRangeEnd(endpoint.KeyspaceGroupIDPrefix())}, "/")
+
+	defaultKGConfigured := false
+	putFn := func(kv *mvccpb.KeyValue) error {
+		group := &endpoint.KeyspaceGroup{}
+		if err := json.Unmarshal(kv.Value, group); err != nil {
+			return errs.ErrJSONUnmarshal.Wrap(err).FastGenWithCause()
+		}
+		kgm.updateKeyspaceGroup(group)
+		if group.ID == mcsutils.DefaultKeyspaceGroupID {
+			defaultKGConfigured = true
+		}
+		return nil
+	}
+	deleteFn := func(kv *mvccpb.KeyValue) error {
+		groupID, err := endpoint.ExtractKeyspaceGroupIDFromPath(string(kv.Key))
+		if err != nil {
+			return err
+		}
+		kgm.deleteKeyspaceGroup(groupID)
+		return nil
+	}
+	postEventFn := func() error {
+		kgm.groupUpdateRetryList.Range(func(key, value interface{}) bool {
+			id, ok := key.(uint64)
```

Comment:
if we use postEventFn, then all accesses to groupUpdateRetryList is in the same goroutine. The sync.Map can changed back to map.